### DISCUSSION TITLE
Adding a script mapper for Identity provider attributes

### DISCUSSION THREAD

**Discussion by laskasn** [keycloak/keycloak#8910] [Ideas]
Currently, keycloak has a script mapper only on the Client side. This type of mapper is very useful when you want to perform a customized mapping, not covered from the other mappings. For instance, if you want to perform a conditional mapping - if this, map using attribA, otherwise map using attribB. or want to have a mapping with perplex concatenations, etc. We have implemented the same "script mapper" for the IdentityProvider side. 
We use this mapper in our installations, with javascript code to perform such complex mapping scenarios, which cannot be done using any of the existing mapper(s).
We think it's a very handy tool and we would like to contribute this to the community.
We submit a pull request with this new feature [here](https://github.com/keycloak/keycloak/pull/7918)

Update: This could also be a keycloak plugin

**Comment by kkmishra233** [comment]
Hi , the PR seems closed without merge. can i expect this feature in any near future. 
I believe the idea was really good and much awaited. don know why keycloak did not bring this by himself.

**Reply by danielFesenmeyer** [reply]
Maybe you like to add a corresponding comment in the associated PR? Or even help to get the PR merged?
